src/pcre2_match.c

```diff
@@ -5937,8 +5937,11 @@ fprintf(stderr, "++ %2ld op=%3d %s\n", Fecode - mb->start_code, *Fecode,
 #ifdef SUPPORT_UNICODE
     if (utf)
       {
-      while (number-- > 0)
+      /* We used to do a simpler `while (number-- > 0)` but that triggers
+      clang's unsigned integer overflow sanitizer. */
+      while (number > 0)
```

Comment:
I think number cannot be 0 (no reverse), so `--number > 0` could work as well.

Reply:
that is what I had [originally](https://github.com/PCRE2Project/pcre2/commit/6c07f89084934af619ac9842b9616b653f8818b0) but I think this looks slightly less ackward.

specially when you add the (what "if somehow number becomes 0 in a future implementation")

Reply:
The number is computed by GET2, so it is uint16_t, and adding +1 could solve the possible 0 value.

Reply:
good idea, but I sill think that just the current version shows intent better and decouples better the concerns that would raise.